nntrainer/tensor/tensor.h

```diff
@@ -504,7 +504,22 @@ class Tensor {
    * @param[in] direction to transpose ex) 0:2:1
    * @retval    Calculated Tensor
    */
-  Tensor transpose(std::string direction) const;
+  Tensor transpose(const std::string &direction) const;
+
+  /**
+   * @brief Transpose Tensor
+   * @param direction to transpose ex) 0:2:1
+   * @param[out] Tensor to saved to
+   * @retval Tensor& reference to the out
+   */
+  Tensor &transpose(const std::string &direction, Tensor &out) const;
+
+  /**
+   * @brief Transpose a tensor and put it back to @a this
+   *
+   * @param direction to transpose ex) 0:2:1
```

Comment:
i highly recommend you to add a note here, that this is NOT an in-place operation. It just returns the result in-place.